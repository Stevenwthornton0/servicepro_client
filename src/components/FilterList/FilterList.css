.filter {
    display: flex;
    flex-direction: column;
    width: 80%;
    margin-left: auto;
    margin-right: auto;
    margin-bottom: 20px;
}

.filterForm {
    display: flex;
    flex-direction: column;
    width: 60%;
}

.filterForm > div {
    margin-right: 16px;
    display: flex;
    flex-direction: column;
}

.filterForm * button {
    font-size: 14px;
    border: 1px solid #bdbdbd;
    border-radius: 4px;
    padding: 3px 6px 3px 6px;
    margin-top: 16px;
    width: 48%;
}

#butContainer {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
}

.filterForm * label {
    margin-bottom: 4px;
}

.filterForm * input {
    padding: 3px 6px 3px 6px;
    font-size: 14px;
    border: 1px solid #bdbdbd;
    border-radius: 4px;
    margin-bottom: 10px;
}

.filter > p {
    margin-top: 0;
    margin-bottom: 16px;
    margin-right: 20px;
    font-size: 18px;
}

@media screen and (min-width: 500px) {
    .filterForm * button:hover {
        cursor: pointer;
        background-color: lightblue;
    }

    .filterForm * button {
        margin-top: 0;
    }

    #butContainer {
        margin-right: 0;
        width: 50%;
    }

    .filter {
        width: 900px;
        display: flex;
        flex-direction: row;
    }

    .filter > p {
        margin-bottom: 0;
    }

    .filterForm {
        display: flex;
        flex-direction: row;
        width: 70%;
    }

    .filterForm * input {
        margin-bottom: 0;
    }

    .filterForm * label {
        margin-bottom: 0;
        margin-right: 4px;
    }

    .filterForm > div {
        display: flex;
        flex-direction: row;
    }
}
