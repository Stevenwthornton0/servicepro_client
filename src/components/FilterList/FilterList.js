import React, { Component } from 'react';
import { States } from '../../Utils/Utils';
import './FilterList.css';

class FilterList extends Component {

    static defaultProps = {
        handleFilter: () => {},
        handleClear: () => {}
    }
    
    render() {
        return (
            <div className='filter'>
                <p className='bold'>Filter by:</p>
                <form className='filterForm' onSubmit={this.props.handleFilter}>
                    <div>
                        <label>City</label>
                        <input
                            required
                            name='city'
                            id='filterList_city'
                        />
                    </div>
                    <div>
                        <label>State</label>
                        <States />
                    </div>
                    <div id='butContainer'>
                        <button type='submit'>Filter</button>
                        <button onClick={this.props.handleClear}>Clear</button>
                    </div>
                </form>
            </div>
        )
    }
}

export default FilterList;