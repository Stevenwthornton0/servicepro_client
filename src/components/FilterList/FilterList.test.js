import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme'
import FilterList from './FilterList';

Enzyme.configure({ adapter: new Adapter() })

describe('FilterList component', () => {
    it('renders without crashing', () => {
        const wrapper = shallow(<FilterList />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})