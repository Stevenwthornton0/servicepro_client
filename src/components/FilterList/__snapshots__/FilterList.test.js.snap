// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterList component renders without crashing 1`] = `
<div
  className="filter"
>
  <p
    className="bold"
  >
    Filter by:
  </p>
  <form
    className="filterForm"
    onSubmit={[Function]}
  >
    <div>
      <label>
        City
      </label>
      <input
        id="filterList_city"
        name="city"
        required={true}
      />
    </div>
    <div>
      <label>
        State
      </label>
      <States />
    </div>
    <div
      id="butContainer"
    >
      <button
        type="submit"
      >
        Filter
      </button>
      <button
        onClick={[Function]}
      >
        Clear
      </button>
    </div>
  </form>
</div>
`;
